### ISSUE TITLE
Ruler extension vs. RISE

### ISSUE THREAD

**Issue by curio77** [ipython-contrib/jupyter_contrib_nbextensions#903]
When running presentations via RISE, the ruler's vertical line is shown at a wrong horizontal position, possibly because it utilizes a font at a different size than the one used to render the code in presentation mode.

Can this be fixed?  If not, would it be possible to have the ruler extension disable itself in RISE presentations?

**Comment by jcb91** [comment]
I would guess your diagnosis is correct & this is something to do with altering font sizes, though I'm not sure. The ruler is a standard codemirror add-on which the nbextension essentially just enables & configures, so I'm a bit surprised it doesn't work correctly already, but at any rate I think a fix should be possible. I've not used RISE, so just to clarify, could you outline minimal steps needed to reproduce the issue?

**Comment by curio77** [comment]
OK, here goes:

1. Install RISE as per [the instructions](https://github.com/damianavila/RISE).
2. Enable the ruler extension. Set the ruler column to a value that's not too high, say 60. (Personally, I have the width set at 72 because that's the limit that RISE will show for code cells on my screen, which is Full-HD, and I'm not sure if that's dependent on the screen resolution.)
3. Create a code cell and fill a line up to the ruler.
4. Launch a presentation via the diagram-like button on the tool bar. Go to the page with the cell if that isn't the first (via spacebar).
5. You'll see that the ruler is not located as expected relative to that line of code previously created.
6. Click the cell, and the ruler will be at the correct location.

**Comment by juhasch** [comment]
I can reproduce it. Can you try to zoom in or zoom out with your browser when in presentation mode ?
This fixes it for me.  I don't see how we could fix this here.

**Comment by curio77** [comment]
Yup, the zoom trick works here too and only needs to be done once per session. :-)

So there's no way to have the ruler extension disable its displaying the ruler in presentation mode (where it's arguably not all that usefuly anyway)?

**Comment by jcb91** [comment]
Is this by any chance related to the `minScale` option, which at the end of https://github.com/damianavila/RISE#configure-your-own-options, it notes "can cause problems with codemirror"?

I'm sure it must be possible to add some css rule to hide rulers in RISE mode, for example, rise appears to [add the `rise-enabled` class to the `body` element](https://github.com/damianavila/RISE/blob/4d187419d6c1ff7a5a6bab81ed8a76668c815830/rise/static/main.js#L210), while codemirror rulers seem to exist within divs that have the `CodeMirror-rulers` class, so I'd guess a simple

```css
body.rise-enabled .CodeMirror-rulers {
  display: none;
}
```

in a `custom.css` ought to do the trick

**Comment by jcb91** [comment]
alternatively, it seems that RISE is doing some slightly hackish things to get code mirror to refresh its sizing (see [rise/static/main.js#L315-L328](https://github.com/damianavila/RISE/blob/4d187419d6c1ff7a5a6bab81ed8a76668c815830/rise/static/main.js#L315-L328), and the accompanying comment at [rise/static/main.js#L491-L492](https://github.com/damianavila/RISE/blob/4d187419d6c1ff7a5a6bab81ed8a76668c815830/rise/static/main.js#L491-L492)), when really it should be waiting for its own transitions to complete before triggering the `cell.code_mirror.refresh()`.

My rulers seem to be in the right place after the refresh, but RISE has added a css rule setting all `pre` elements to `width: 90%`, which applies to CodeMirror lines, and cuts the ends off :frowning_face:
